src/test/java/hw2/base/BaseTest.java

```diff
@@ -0,0 +1,146 @@
+        Assert.assertEquals(images.size(), 4);
+        for (WebElement img : images) {
+            Assert.assertTrue(img.isDisplayed());
+        }
+    }
+
+    public void checkTextBelowIcons() {
+        List<WebElement> texts = driver.findElements(By.className(textsOnHomePage));
+        assertEquals(texts.size(), 4);
+        for (WebElement text : texts) {
+            assertTrue(text.isDisplayed());
+        }
+    }
+
+    public void assertFrameButton() {
+        WebElement frame = driver.findElement(By.id(buttonWithNameFrame));
+        boolean existedFrame = frame.isEnabled();
+        Assert.assertTrue(existedFrame);
+    }
+
+    public void switchToTheFrameAndCheck() {
+        WebElement frame = driver.findElement(By.id(buttonWithNameFrame));
+        driver.switchTo().frame(frame);
+        WebElement iFrameButton = driver.findElement(By.id(buttonFrame));
+        assertTrue(iFrameButton.isDisplayed());
+    }
+
+    public void switchToOriginalWindowBack() {
+        driver.switchTo().defaultContent();
+    }
+
+    public void assertItemsOnTheLeftSection() {
+        WebElement items = driver.findElement(By.cssSelector(leftSection));
+        assertTrue(items.isDisplayed());
+        assertEquals(items.getText(), menuButtonsOnTheLeft);
+    }
+
+    //ex2 steps
+    public void openDifferentElementsPage() {
+        WebElement element = driver.findElement((By.xpath(pageService)));
+        element.click();
+        driver.findElement((By.xpath(differentElementPage)));
+        element.click();
+        driver.get(URL_DIFF_ELEMENT_PAGES);
+    }
+
+    public void selectWaterCheckbox() {
+        driver.findElement(By.cssSelector(water)).click();
+    }
+
+    public void selectWindCheckbox() {
+        driver.findElement(By.cssSelector(wind)).click();
+    }
+
+    public void selectRadio() {
+        driver.findElement(By.xpath(selen)).click();
+    }
+
+    public void selectDropdown() {
+        driver.findElement((By.xpath(uuiFormElement))).click();
+        driver.findElement((By.xpath(yellow))).click();
+    }
+
+    public void assertCheckboxes() {
+        List<WebElement> logList = driver.findElements(By.cssSelector(panelBodyListLogs));
+        assertTrue(logList.get(2).getText().contains("Wind: condition changed to true"));
+        assertTrue(logList.get(3).getText().contains("Water: condition changed to true"));
+    }
+
+    public void assertRadio() {
+        List<WebElement> logsList = driver.findElements(By.cssSelector(panelBodyListLogs));
+        assertTrue(logsList.get(1).getText().contains("metal: value changed to Selen"));
+    }
+
+    public void assertInDropdown() {
+        List<WebElement> logsList = driver.findElements(By.cssSelector(panelBodyListLogs));
+        assertTrue(logsList.get(0).getText().contains("Colors: value changed to Yellow"));
+    }
+
+    public static final String BASE_URL = "https://jdi-testing.github.io/jdi-light/index.html";
```

Comment:
1) move url to constants
2) you should have 1 url - common url. No url-s anymore. All navigation start from base url and perform via silenium actions!!!!

Moreover you cheating in this method. If I remove last line of method:
```
    public void openDifferentElementsPage() {
        WebElement element = driver.findElement((By.xpath(pageService)));
        element.click();
        driver.findElement((By.xpath(differentElementPage)));
        element.click();
        driver.get(URL_DIFF_ELEMENT_PAGES);
    }
```

test will fail because test not go for different element page :(